File: src/121BestTimeToBuyAndSellStock/121BestTimeToBuyAndSellStock.go
package main

func maxProfit(prices []int) int {
	n := len(prices)
	local := 0
	global := 0
	for i := 1; i < n; i++ {
		diff := local + prices[i] - prices[i-1]
		if diff < 0 {
			local = 0
		} else {
			local = diff
		}
		if local > global {
			global = local
		}
	}
	return global
}

func main() {

}

File: src/121BestTimeToBuyAndSellStock/121BestTimeToBuyAndSellStock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut local = 0;
        let mut global = 0;
        for i in 0..(prices.len() as i32)-1 {
            local = std::cmp::max(local+prices[i as usize +1]-prices[i as usize],0);
            global= std::cmp::max(local, global);
        }
        global
    }
}
